### ISSUE TITLE
loss is displayed incorrectly?

### ISSUE THREAD

**Issue by iMountTai** [meta-llama/llama-recipes#193]
### System Info

pytorch-nightly
cu118
TeslaP40

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

Do you have a speed test sample? FSDP+PEFT runs too slowly. And loss is displayed incorrectly? The loss shown is the result of dividing gradient_accumulation_steps.
```
enable_fsdp
gradient_accumulation_steps=144
use_fp16=true
batch_size_training=1
use_peft
peft_method="lora"
```
Transformers DDP is three times as fast as FSDP+PEFT

### Error logs

![image](https://github.com/facebookresearch/llama-recipes/assets/35353688/0271858f-c0b6-41b2-b027-db709926d7df)

### Expected behavior

As above.

**Comment by mreso** [comment]
Hi @iMountTai, you're right, we display the loss/gradient_accumulation_steps as we do not save the batch loss for siplaying. Would you be interested in providing a pr that fixes this by doing a loss.detach().float() an save the loss in a variable that gets used for the pbar update?

Regarding the speed of FSPD+PEFT. What is you setup (model size, how many gpus) etc? Do you also mean DDP+PEFT?

**Comment by HamidShojanazeri** [comment]
> Transformers DDP is three times as fast as FSDP+PEFT
@iMountTai note that FSDP has 3x more communication overhead compared to DDP. It basically shards the model parameters, gradients and optimizer states. So you would want to use it only if necessary, it unlock the ability to train models which are not fitting into one GPU.

Saying so, in FSDP we have other sharding strategies, can be set [here](https://github.com/facebookresearch/llama-recipes/blob/main/src/llama_recipes/configs/fsdp.py#L13) that resemble more to DDP, `SHARD_GRAD_OP` which is Zero2 basically keep the model parameters unsharded after first forward and only shard gradients and optimizer states, `NO_SHARD` is basically DDP.

Please feel free to set as suits you and your network speed, less communication as in zero2 , SHARD_GRAD_OP` or `NO_SHARD` are more sutiable for slower networks for example when you dont have EFA.

**Comment by HamidShojanazeri** [comment]
@iMountTai Also make sure your GPU support BF16 if you are using that otherwise the slow down is significant and it wont throw error.
